usecase/call_rpc.go

```diff
@@ -46,21 +50,30 @@ func (e *gRPCError) Code() ErrorCode {
 // the request to the gRPC server and decodes the response body to res.
 // Note that req and res must be JSON-decodable structs. The output is written to w.
 func CallRPC(ctx context.Context, w io.Writer, rpcName string) error {
-	return dm.CallRPC(ctx, w, rpcName, dm.filler)
+	return dm.CallRPC(ctx, w, rpcName, false, dm.filler)
 }
-func (m *dependencyManager) CallRPC(ctx context.Context, w io.Writer, rpcName string, filler fill.Filler) error {
+func (m *dependencyManager) CallRPC(ctx context.Context, w io.Writer, rpcName string, rerunPrevious bool, filler fill.Filler) error {
 	fqsn := proto.FullyQualifiedServiceName(m.state.selectedPackage, m.state.selectedService)
 	rpc, err := m.spec.RPC(fqsn, rpcName)
```

Comment:
We should return an error if `rerunPrevious` is specified for a ClientStreaming/BidiStreaming RPC. For example,

``` go
if rerunPrevious && rpc.IsClientStreaming {
	return errors.New("cannot rerun previous RPC as client/bidi streaming RPCs are not supported")
}
```
```

Reply:
This happens in the `getPreviousRPCRequest()`. Didn't want to pollute the `callRPC()` with the responsibility of checking if `isClientStreaming()`. However, as an optimization, we'll save few compute cycles by failing fast. Do you see adding it up top as an advantage?

Reply:
Evans has hung up in the previous implementation when calling a BidiStreaming with `--repeat` option.

```
$ evans -s -r repl --package grpc.reflection.v1alpha --service ServerReflection
grpc.reflection.v1alpha.ServerReflection@127.0.0.1:50051> call --repeat ServerReflectionInfo 
# Hung up!
```

The root cause is the `errgroup` which is used for asynchronous processing in the request/response loop (L148-247). We can still keep code without fast-failing, however, it's difficult to keep simplicity.
I think fast-failing is the simplest and most maintainable. What do you think?

Reply:
Agreed. Failing fast seems reasonable here.